### PR TITLE
Test keys set to empty strings

### PR DESCRIPTION
**Pull request by ppalaga**
https://github.com/editorconfig/editorconfig/issues/496

### CODE REVIEW

File: parser/CMakeLists.txt [hunk 1]
```diff
@@ -129,3 +129,6 @@ new_ec_test(empty_editorconfig_file empty.in test4 "^[ \t\n\r]*$")
 new_ec_test(newlines_only_editorconfig_file newlines_only.in test4 "^[ \t\n\r]*$")
 new_ec_test(comments_only_editorconfig_file comments_only.in test4 "^[ \t\n\r]*$")
 new_ec_test(comments_and_newlines_editorconfig_file comments_and_newlines.in test4 "^[ \t\n\r]*$")
+
+# A key set to empty string
+new_ec_test(key_set_to_empty_string empty_values.in test1 "^key1=[\n\r]*key2=[\n\r]*key3=[\n\r]*$")
```
Comment (cxw42) [inline parser/CMakeLists.txt:134]: I think this should be
```suggestion
new_ec_test(key_set_to_empty_string empty_values.in test1 "^key1=[\n\r]+key2=[\n\r]+key3=[\n\r]+$")
```
so we don't accept, e.g., `key1=key2` as passing this test.

**Review by cxw42 — CHANGES_REQUESTED** [review]

**Comment by cxw42** [comment]
Would it also be worth adding a testcase to `parser/whitespace.in` like
```
key= value with whitespace inside  $
``` (`$` = end of line)?  That way we would expressly show the behaviour of preserving whitespace inside values.  Currently we test that implicitly in comments.in, test `semicolon_or_hash_in_property`, but we don't have a test in the `whitespace` group for it.

File: parser/CMakeLists.txt (hunk 1 above)
Comment (ppalaga) [reply parser/CMakeLists.txt]: Thanks fixed in 9877f04

**Comment by ppalaga** [comment]
> Would it also be worth adding a testcase to `parser/whitespace.in` like
> 
> ```
> key= value with whitespace inside  $
> ```
> 
> (`$` = end of line)? That way we would expressly show the behaviour of preserving whitespace inside values. Currently we test that implicitly in comments.in, test `semicolon_or_hash_in_property`, but we don't have a test in the `whitespace` group for it.

Done in 9877f04

Could you please review again?

**Comment by ppalaga** [comment]
Any chance to re-review @cxw42 or anybody else?

File: parser/CMakeLists.txt [hunk 2]
```diff
@@ -129,3 +133,6 @@ new_ec_test(empty_editorconfig_file empty.in test4 "^[ \t\n\r]*$")
 new_ec_test(newlines_only_editorconfig_file newlines_only.in test4 "^[ \t\n\r]*$")
 new_ec_test(comments_only_editorconfig_file comments_only.in test4 "^[ \t\n\r]*$")
 new_ec_test(comments_and_newlines_editorconfig_file comments_and_newlines.in test4 "^[ \t\n\r]*$")
+
+# A key set to empty string
```
Comment (xuhdev) [inline parser/CMakeLists.txt:137]: Do you mean "value" set to empty string? This is the only confusing spot, otherwise the PR looks good.

**Review by cxw42 — APPROVED** [review]
LGTM once that comment is fixed!

**Comment by ppalaga** [comment]
a665d19: `s/key/value/` in the comment as requested by reviewers

File: parser/empty_values.in [hunk 3]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[test1]
+; A key set to empty string
```
Comment (xuhdev) [inline parser/empty_values.in:4]: Sorry missed this one also :)

Comment (xuhdev) [reply parser/empty_values.in]: (I took the liberty to fix this)

Comment (ppalaga) [reply parser/empty_values.in]: Thanks a lot!

**Pull request merged**